internal/providers/testing/provider_mock.go

```diff
@@ -26,6 +26,9 @@ var _ providers.Interface = (*MockProvider)(nil)
 type MockProvider struct {
 	sync.Mutex
 
+	// NoLock is used to disable locking in the mock provider.
+	NoLock bool
```

Comment:
sorry, this didn't occur to me before. Would it work just to have a test that uses two (or more) separate provider instances so they don't share the mutex under the hood anyway? This would be instead of giving the option to disable the locking.

Reply:
The test runs share the same configuration, and I am not sure how would then differentiate the instance they should use.